Makefile

```diff
@@ -12,6 +12,19 @@ aw-server:
 aw-webui:
 	make -C ./aw-webui build
 
+install-git-hooks:
+	@printf "make pre-commit\n" > .git/hooks/pre-commit
+	@printf "make pre-push\n" > .git/hooks/pre-push
+	@chmod +x .git/hooks/pre-commit .git/hooks/pre-push
+	@printf "Hooks installed\n"
+
+pre-commit:
+	@cargo fmt -- --check || (printf "Error: Run cargo fmt before committing\n"; exit 1)
+
+pre-push: pre-commit
+	@cargo clippy || (printf "Error: Clippy reported error(s)\n"; exit 1)
+	@cargo test || (printf "Error: Some test(s) failed\n"; exit 1)
```

Comment:
I don't think we should run this in pre-push as running all tests takes a long time.
It will be verified in CI as well so you can't skip it anyway.

Reply:
In my experience running aw-server-rust's texts usually takes 10-20 seconds. Fair point about the CI though.

Reply:
The tests execute fast yes, but when you count in compilation time it is very slow. Sometimes you only want to modify a single crate and not the whole project, in that case it might take 5 minutes extra with this commit hook.

Reply:
The default hook installed from the Makefile should definitely skip this if it takes whole 5 minutes, and CI checks it anyways. I personally prefer always running tests or modified code before pushing, but I can keep my local git hooks like that without affecting others.

Reply:
I'd guess long build times mean building for release, either by the make build task (which also builds aw-webui which is quite slow if not ready). 

I run `cargo check``cargo run` and `cargo test` on my machine and it is quite fast after the first build.

Reply:
@xylix: Please note that currently this PR only enables format checking for commits; clippy and tests are only ran pre-push. Still, relying only on CI is certainly possible. However, when working with feature branches or on my own fork, requiring tests to pass before pushing is quite useful.

Reply:
I agree that it's useful, but I don't think it's nice to enforce a workflow on users. Especially when it can potentially take a lot of time.